### PR TITLE
refactor: Copy and refactor nav

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it copies nav from fashion branch and refactores it.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/lib/helpers.ts [hunk 1]
```diff
@@ -1,65 +1,9 @@
-import type { MenuItem } from "@nimara/domain/objects/Menu";
-import { loggingService } from "@nimara/infrastructure/logging/service";
+export const isValidJson = (value: string) => {
+  try {
+    JSON.parse(value);
 
-interface Paths {
```
Comment (karolkarolka) [inline apps/storefront/src/lib/helpers.ts:4]: what has happend to the deleted lines in this file? Aren't they or weren't used in Nimara anymore?

File: apps/storefront/src/components/mobile-navigation.tsx [hunk 2]
```diff
@@ -23,33 +21,21 @@ export const MobileNavigation = ({
...
-                </ul>
-              ) : null}
-            </Link>
-          ) : (
-            <a href={item.url as string} target="_blank">
-              {item.name}
-            </a>
-          )}
+        <li key={item.id} className="p-2 text-stone-500">
+          <Link href={item.url} onClick={() => handleMenuItemClick()}>
```
Comment (karolkarolka) [inline apps/storefront/src/components/mobile-navigation.tsx:25]:
```suggestion
          <Link href={item.url} onClick={() => onMenuItemClick(true)}>
```

File: apps/storefront/src/components/mobile-navigation.tsx [hunk 3]
```diff
@@ -23,33 +21,21 @@ export const MobileNavigation = ({
...
+            {item.label}
+          </Link>
+          {item.children?.length ? (
+            <ul className="mt-2 pl-6">
+              {item.children.map((child) => (
+                <li key={child.id} className="py-1 pl-2 text-stone-700">
+                  <Link href={child.url} onClick={() => handleMenuItemClick()}>
```
Comment (karolkarolka) [inline apps/storefront/src/components/mobile-navigation.tsx:32]:
```suggestion
                  <Link href={child.url} onClick={() =>onMenuItemClick(true)}>
```

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 4]
```diff
@@ -1,62 +1,123 @@
...
+                                  <p className="py-1">{child.description}</p>
+                                )}
+                              </div>
+                            </Link>
+                          ))
+                      : null}
+                  </div>
+
+                  <div className="col-span-4 grid grid-cols-3 gap-3">
+                    {item.children?.length
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:80]:
```suggestion
                    {!!item.children?.length &&
```

File: apps/storefront/src/components/mobile-navigation.tsx [hunk 5]
```diff
@@ -23,33 +21,21 @@ export const MobileNavigation = ({
...
-          ) : (
-            <a href={item.url as string} target="_blank">
-              {item.name}
-            </a>
-          )}
+        <li key={item.id} className="p-2 text-stone-500">
+          <Link href={item.url} onClick={() => handleMenuItemClick()}>
+            {item.label}
+          </Link>
+          {item.children?.length ? (
```
Comment (karolkarolka) [inline apps/storefront/src/components/mobile-navigation.tsx:28]:
```suggestion
          {!!item.children?.length && (
```

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 6]
```diff
@@ -1,62 +1,123 @@
...
-                      paths.search.asPath({
-                        query: {
-                          category: item.category.slug,
-                        },
-                      }),
-                    )
-                  : undefined
-              }
+            <Link
+              href={item.url || "#"}
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:37]:
```suggestion
              href={item.url}
```

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 7]
```diff
@@ -1,62 +1,123 @@
...
-                          {child.name ||
-                            child.collection?.name ||
-                            child.category?.name}
-                        </div>
-                      </Link>
-                    </NavigationMenuLink>
-                  ))}
+                <div className="grid w-full grid-cols-6 p-6">
+                  <div className="col-span-2 flex flex-col gap-3 pr-6">
+                    {item.children?.length
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:48]:
```suggestion
                    {!!item.children?.length &&
```

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 8]
```diff
@@ -1,62 +1,123 @@
...
+                      ? item.children
+                          ?.filter((child) => !child.collectionImageUrl)
+                          .map((child) => (
+                            <Link
+                              key={child.id}
+                              href={child.url}
+                              className="group block space-y-1 rounded-md p-3 hover:bg-accent"
+                              onClick={() => {
+                                setCurrentMenuItem("");
+                              }}
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:58]:
```suggestion
                              onClick={() => setCurrentMenuItem("")}
```
why the onClick event should set the component's state as `menu item = empty string`?

File: packages/infrastructure/src/lib/serializers/cms-menu.ts [hunk 9]
```diff
@@ -1,86 +1,147 @@
...
+    id: child.meta.id.toString(),
+    label: child.name,
+    url:
+      child.url ||
+      createMenuItemUrl(
+        child.category_slug ? { slug: child.category_slug } : null,
+        child.collection_slug ? { slug: child.collection_slug } : null,
+        child.page_slug ? { slug: child.page_slug } : null,
+      ) ||
+      "#",
```
Comment (karolkarolka) [inline packages/infrastructure/src/lib/serializers/cms-menu.ts:92]: why #?

File: packages/infrastructure/src/public/butter-cms/cms-menu/infrastructure/cms-menu-get-infra.ts [hunk 10]
```diff
@@ -31,15 +30,48 @@ export const butterCMSMenuGetInfra =
...
+
+    console.log(
+      "<<<< SUBMENU",
+      JSON.stringify(
+        submenu?.data?.data?.navigation_menu_item_second_level,
+        null,
+        2,
+      ),
+    );
+
```
Comment (karolkarolka) [inline packages/infrastructure/src/public/butter-cms/cms-menu/infrastructure/cms-menu-get-infra.ts:59]:
```suggestion
```

**Review by karolkarolka — CHANGES_REQUESTED** [review]

File: apps/storefront/src/lib/helpers.ts (hunk 1 above)
Comment (soniaklimas) [reply apps/storefront/src/lib/helpers.ts:4]: All deleted functions are not used anymore in Nimara. Refactor of navigation simplified creating urls and most of that functions were related to that.

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx (hunk 8 above)
Comment (soniaklimas) [reply apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:58]: It manually closes menu - if you click on given menu item menu should not be visible. It is workaround to achieve that.

File: packages/infrastructure/src/lib/serializers/cms-menu.ts (hunk 9 above)
Comment (karolkarolka) [reply packages/infrastructure/src/lib/serializers/cms-menu.ts:92]: Yeah but all of the elements in the menu should be links so if someone won't set links via CMS the storefront should be notified so either make URLs as required fields in types or make a condition `item.url && <Link`. Setting the hash as the link value when it wasn't set in CMS doesn't seem right to me. The empty hash will navigate to the top of the page and be weirdly visible in URL

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 11]
```diff
@@ -1,65 +1,120 @@
...
+                  <div className="col-span-2 flex flex-col gap-3 pr-6">
+                    {!!item.children?.length &&
+                      item.children
+                        ?.filter((child) => !child.collectionImageUrl)
+                        .map((child) => (
+                          <Link
+                            key={child.id}
+                            href={child.url}
+                            className="group block space-y-1 rounded-md p-3 hover:bg-accent"
+                            // onClick={() => setCurrentMenuItem("")}
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:56]: to remove?

File: apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx [hunk 12]
```diff
@@ -1,65 +1,120 @@
...
+                          <Link
+                            key={child.id}
+                            href={child.url}
+                            className="group relative min-h-[270px] overflow-hidden rounded-lg bg-accent"
+                            onClick={() => setCurrentMenuItem("")}
+                          >
+                            <div
+                              className="h-1/2 bg-cover bg-center"
+                              style={{
+                                backgroundImage: `url(${child?.collectionImageUrl})`,
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:91]: Can you check if the URLs uploaded to the CMS are handled by their CDNs and are returned as optimized images? The problem with setting the image as a background of the five is that the images won't be optimized and might cause the slow render issues

Comment (soniaklimas) [reply apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:91]: Images (from Butter CMS and Saleor as well) are served in WebP format. They are also delivered via a CDN (CloudFront) with caching enabled. Is that enough?

Comment (karolkarolka) [reply apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx:90]: Yes. If the images will be served in WebP or AVIF instead of raw/large PNGs, JPEGs, there would be no need to switch from background image to next/image

**Review by karolkarolka — APPROVED** [review]
